WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/usecases/GetPublishButtonLabelUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package org.wordpress.android.ui.posts.prepublishing.home.usecases
+
+import android.text.TextUtils
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.ui.posts.EditPostRepository
+import org.wordpress.android.ui.posts.PostUtils
+import javax.inject.Inject
+
+typealias StringResourceId = Int
+
+class GetPublishButtonLabelUseCase @Inject constructor() {
+    fun getLabel(editPostRepository: EditPostRepository): StringResourceId {
+        val dateCreated = editPostRepository.dateCreated
+        val status = editPostRepository.status
+        return when {
+            !TextUtils.isEmpty(dateCreated) -> {
+                when {
+                    status == PostStatus.SCHEDULED -> R.string.prepublishing_nudges_home_schedule_button
+                    status == PostStatus.PUBLISHED || status == PostStatus.PRIVATE ->
+                        R.string.prepublishing_nudges_home_publish_button
+                    editPostRepository.isLocalDraft -> R.string.prepublishing_nudges_home_publish_button
+                    PostUtils.isPublishDateInTheFuture(editPostRepository.dateCreated) ->
```

Comment:
We might want to consider using injectable `PostUtilsWrapper`. 

I'd consider the following
1. Add `isPublishDateInTheFuture` into the `PostUtilsWrapper`
2. Use the wrapper instead of the static PostUtils in the GetPublishButtonLabelUseCase
3. Inject DateProvider into the wrapper
4. Create a version of `isPublishDateInTheFuture` in PostUtils which would accept even the currentDate and call it from the wrapper
5. Remove the non-deterministic behavior from the unit tests -> mock even the "currentDate" so we don't need to use `new Date()` and we can hardcode all the dates in the tests.

Let me know if it's clear or I should elaborate more.

Reply:
Nice! Thanks for spotting this. Your explanation is quite clear and I do agree with you about making the test more deterministic. I will implement the changes and let you review 🙏